client/src/components/reader/Section/Pagination.js

```diff
@@ -6,22 +6,22 @@ export default class Pagination extends Component {
   static propTypes = {
     textId: PropTypes.string,
     sectionId: PropTypes.string,
-    textSections: PropTypes.array
+    spine: PropTypes.array
   };
 
   getSiblingSection(id, shift) {
     let siblingSection = false;
-    this.props.textSections.forEach((section, i) => {
-      if (section.id === id && this.props.textSections[i + shift]) {
-        siblingSection = this.props.textSections[i + shift];
+    this.props.spine.forEach((spineId, i) => {
```

Comment:
ForEach seems like an inefficient approach here. Seems like you can just use findIndex (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) to get the index of the current section, then add shift to that to get the section you want.